docs/feature-store/transformations.md

```diff
@@ -189,4 +189,83 @@ quotes_set.graph.add_step("MyMap", "multi", after="filter", multiplier=3)
 ```
 
 This uses the `add_step` function of the graph to add a step called `multi` utilizing `MyMap` after the `filter` step 
-that was added previously. The class is initialized with a multiplier of 3.
\ No newline at end of file
+that was added previously. The class is initialized with a multiplier of 3.
+
+## Supporting multiple engines
+
+MLRun supports multiple processing engines for executing graphs. These engines differ in the way they invoke graph
+steps. When implementing custom transformations, the code has to support all engines which are expected to run it. 
+
+```{admonition} Note
+The vast majority of MLRun's built-in transformations support all engines. The support matrix is available 
+[here](../serving/available-steps.html#data-transformations).
+```
+
+The following are the main differences between transformation steps executing on different engines:
+
+* `storey` - the step receives a single event (either as a dictionary or as an Event object, depending on whether 
+  `full_event` is configured for the step). The step is expected to process the event and return the modified event.
+* `spark` - the step receives a Spark dataframe object. Steps are expected to add their processing and calculations to 
+  the dataframe (either in-place or not) and return the resulting dataframe without materializing the data. 
+* `pandas` - the step receives a Pandas dataframe, and processes it. It then returns the dataframe.
```

Comment:
* `pandas` - the step receives a Pandas dataframe, processes it, and returns the dataframe.